tools/wptserve/tests/functional/test_pipes.py

```diff
@@ -43,11 +44,11 @@ def test_semicolon(self):
         self.assertEqual(resp.info()["Refresh"], "3;url=http://example.com")
 
     def test_escape_comma(self):
-        resp = self.request("/document.txt", query="pipe=header(Expires,Thu\,%2014%20Aug%201986%2018:00:00%20GMT)")
+        resp = self.request("/document.txt", query=r"pipe=header(Expires,Thu\,%2014%20Aug%201986%2018:00:00%20GMT)")
```

Comment:
Sorry, I missed these changes before. Why are these changes required?

Reply:
This is because we have `\` in the string. Python 3 interprets string literals as Unicode strings. In this case Py3 gives warning like `DeprecationWarning: invalid escape sequence \`. Using a raw string here is to make sure that `\` is interpreted as a plain `\` rather than escape sequences. It is also recommended in [re](https://docs.python.org/3/library/re.html) to use raw string for regular expressions.

Reply:
Where is the regexp in this test? This test appears to send a request to the wptserve server started by the `TestUsingServer` class (in `tools/wptserve/tests/functional/base.py`), and asserts that the return value has the right 'Expires' value (subbed in via a `pipe`). More specifically, from what I can see the `query` parameter is:

1. Passed into `request`
1. Then passed into `self.abs_url`
1. Then passed into `urlunsplit` (aka `six.moves.urllib.parse.urlunsplit`)
1. The result of `urlunsplit` is then passed to the `Request` class constructor
1. The `Request` class constructor passed it to `BaseRequest` (aka `six.moves.urllib.request.Request`).

Which part of this uses a regexp? (Or did I miss something).

It seems like adding `r` here **might** be covering up something that may be a bug, or it may be fine, and I'm not confident currently which it is! :)

Reply:
Ah sorry! There is no regexp call here. I have messed it up with the change of "assertRegex".

Reply:
> This is because we have `\` in the string. Python 3 interprets string literals as Unicode strings. In this case Py3 gives warning like `DeprecationWarning: invalid escape sequence \`. Using a raw string here is to make sure that `\` is interpreted as a plain `\` rather than escape sequences. 

This is the reason we use raw string here.

Reply:
Ok. I think I understand; the `regex` part wasn't actually relevant, but this is still using an unescaped backslash in a string which is a DeprecationWarning in Py3+. The reason being that if our comma was actually a valid escape sequence, we would end up with a different string:

```
>>> x = "\ndsf"
>>> y = r"\ndsf"
>>> x
'\ndsf'
>>> y
'\\ndsf'
```

Ok, this LGTM.